pkg/metrics/producers/scheduledcapacity/schedules.go

```diff
@@ -0,0 +1,500 @@
+				singleSkipChecker := strings.Split(val, "/")
+				if len(singleSkipChecker) > 2 {
+					return s, fmt.Errorf("cannot specify more than one skip operator per field")
+				}
+
+				// if the field is */int
+				if len(singleSkipChecker) == 2 {
+					moduli, err := strconv.Atoi(singleSkipChecker[1])
+					if err != nil {
+						return s, err
+					}
+					for bitToSet := 0 ; bitToSet < fieldTypeToAsteriskIndex[i] ; bitToSet += 1 {
+						if bitToSet % moduli == 0 {
+							s.SetBitTrue(bitToSet, i)
+						}
+					}
+				} else {
+					// If wildcard, fill in wildcard index
+					if val == "*" {
+						s.SetBitTrue(i, fieldTypeToAsteriskIndex[i])
+					} else {
+						singleValInt, err := strconv.Atoi(val)
+						if err != nil {
+							return s, err
+						}
+						if rangeValidators[i](singleValInt) {
+							return s, fmt.Errorf("value in %s field is not in valid range", errorNames[i])
+						}
+					}
+				}
+			}
+			// true if current parsing element is a range of values
+			if len(dashChecker) == 2 {
+				// check to see if the skip operator is in play
+				skipCheck := strings.Split(dashChecker[1], "/")
+				if len(skipCheck) > 2 {
+					return s, fmt.Errorf("a crontab field cannot have multiple skip operators in one element")
+				}
+
+				// parse the start of the range
+				rangeStartInt, err := strconv.Atoi(dashChecker[0])
+				if err != nil {
+					return s, err
+				}
+
+				// declare end of range here, as it needs to go through more parsing
+				var rangeEndInt int
+				moduli := 1
+
+				// If current parsing element has a skip operator
+				if len(skipCheck) == 2 {
+					rangeEndInt, err = strconv.Atoi(skipCheck[0])
+					if err != nil {
+						return s, err
+					}
+					moduli, err = strconv.Atoi(skipCheck[1])
+					if err != nil {
+						return s, err
+					}
+				}
+				// If current parsing element has no skip operator
+				if len(skipCheck) == 1 {
+					rangeEndInt, err = strconv.Atoi(dashChecker[1])
+					if err != nil {
+						return s, err
+					}
+				}
+
+				// If start or end of range is invalid given the field type
+				if rangeValidators[i](rangeStartInt) || rangeValidators[i](rangeEndInt) {
+					return s, fmt.Errorf("value specified in %s field is an invalid range", errorNames[i])
+				}
+				if rangeStartInt > rangeEndInt {
+					return s, fmt.Errorf("range of values specified in %s field have a greater start than end", errorNames[i])
+				}
+
+				for bitToSet := rangeStartInt; bitToSet <= rangeEndInt; bitToSet++ {
+					// if there is no skip operator, this is always true
+					if bitToSet % moduli == rangeStartInt % moduli {
+						s.SetBitTrue(i, bitToSet)
+					}
+				}
+			}
+		}
+	}
+	return s, nil
+}
+
+func findMostRecentMatches(spec ScheduleSpec, startTime time.Time, endTime time.Time) (*apis.VolatileTime, *apis.VolatileTime, error) {
+	iterBack := &iterativeSchedule{
+		Year:   time.Now().Year(),
+		Minute: time.Now().Minute(),
+		Hour:   time.Now().Hour(),
+		Dom:    time.Now().Day(),
```

Comment:
Not sure the `dom` abbreviation is worth it... I keep thinking of JavaScript DOMs.